### ISSUE TITLE
Issue with karma-runner and the has-binary dep

### ISSUE THREAD

**Issue by kentor** [socketio/socket.io#2198]
Also filed in the karma-runner repo: https://github.com/karma-runner/karma/issues/1534

I have some jasmine tests where the describe block is named `toJSON`. Somehow we get to this line of code in the `has-binary` module that this project depends on:

``` javascript
      if (obj.toJSON) {
        obj = obj.toJSON();
      }
```

`toJSON` isn't a function so this just blows up.

Also, by the way, seems like that code is included twice in the `socket.io.js` bundle.

**Comment by kentor** [comment]
ah sorry I think this should have gone to the socket.io-client repo

**Comment by RobbinHabermehl** [comment]
I can confirm this issue, changing the `describe` block name resolves it.

**Comment by wesleycho** [comment]
`has-binary` should probably check if it is a function - @rauchg this looks like it is your library, and doesn't seem to have a repository on GitHub, so one cannot file a PR to fix it. Can you patch it up with a simple function check and release a new version?

**Comment by darrachequesne** [comment]
It seems it was fixed in version `0.1.7` ([has-binary](https://github.com/socketio/has-binary))

Yet `engine.io-parser` still relies on [`0.1.6`](https://github.com/socketio/engine.io-parser/blob/master/package.json#L18).

**Comment by darrachequesne** [comment]
@kentor @wesleycho was this fixed by `1.7.2`?

**Comment by kentor** [comment]
yes i believe this is fixed
